src/storage/src/mem_table.rs

```diff
@@ -89,27 +106,43 @@ impl MemTable {
     }
 
     pub fn delete(&mut self, pk: Bytes, old_value: Bytes) -> Result<()> {
-        let entry = self.buffer.entry(pk);
+        if !self.is_consistent_op {
```

Comment:
What does `is_consistent_op` mean here? 👀

Reply:
This is simply the `!disable_sanity_check`. I change the term `disable_sanity_check` to a property of a state table to make its semantic clearer.

I have moved some memtable and state table related changes to https://github.com/risingwavelabs/risingwave/pull/7579 to decrease the size of this PR. The `is_consistent_op` will be introduced in that PR.